src/main/java/com/google/gcp_variant_transforms/library/VariantToBqUtils.java

```diff
@@ -0,0 +1,241 @@
+    return dealWithMissingFieldValue(variantContext.getID());
+  }
+
+  public static List<TableRow> addAlternates(VariantContext variantContext,
+                                             Map<String, Integer> alleleIndexingMap) {
+    List<TableRow> altMetadata = new ArrayList<>();
+    List<Allele> altAlleles = variantContext.getAlternateAlleles();
+    // if field value is '.', alternateAllelesList will be empty
+    if (altAlleles.isEmpty()) {
+      TableRow subRow = new TableRow();
+      subRow.set(ColumnKeyConstants.ALTERNATE_BASES_ALT, null);
+      altMetadata.add(subRow);
+    } else {
+      for (int i = 0; i < altAlleles.size(); ++i) {
+        TableRow subRow = new TableRow();
+        String altBase = altAlleles.get(i).getDisplayString();
+        alleleIndexingMap.put(altBase, i + 1);  // the alt allele index is 1-based
+        subRow.set(ColumnKeyConstants.ALTERNATE_BASES_ALT, altBase);
+        altMetadata.add(subRow);
+      }
+    }
+    return altMetadata;
+  }
+
+  /**
+   * Add variant Info field into BQ table row.
+   * If the info field number in the VCF header is `A`, add it to the ALT sub field.
+   */
+  public static void addInfo(TableRow row, VariantContext variantContext,
+                             List<TableRow> altMetadata, VCFHeader vcfHeader) {
+    for (Map.Entry<String, Object> entry : variantContext.getAttributes().entrySet()) {
+      String attrName = entry.getKey();
+      Object value = entry.getValue();
+      VCFInfoHeaderLine infoMetadata = vcfHeader.getInfoHeaderLine(attrName);
+      VCFHeaderLineType infoType = infoMetadata.getType();
+      VCFHeaderLineCount infoCount = infoMetadata.getCountType();
+      if (infoCount == VCFHeaderLineCount.A) {
+        // put this info into ALT field
+        splitAlternateAlleleInfoFields(attrName, value, altMetadata, infoType);
+      } else {
+        row.set(attrName, convertToDefinedType(value, infoType));
+      }
+    }
+  }
+
+  public static List<TableRow> addCalls(GenotypesContext genotypes, VCFHeader vcfHeader,
+                                        Map<String, Integer> alleleIndexingMap) {
+    List<TableRow> callRows = new ArrayList<>();
+    for (Genotype genotype : genotypes) {
+      TableRow curRow = new TableRow();
+      curRow.set(ColumnKeyConstants.CALLS_NAME, genotype.getSampleName());
+      addInfoAndPhaseSet(curRow, genotype, vcfHeader);
+      addGenotypes(curRow, genotype.getAlleles(), alleleIndexingMap);
+      callRows.add(curRow);
+    }
+    return callRows;
+  }
+
+  /**
+   * Convert the value to the type defined in the metadata
+   * If the value is an instance of List, iterate the list
+   * and call convertSingleObjectToDefinedType() to convert every value
+   */
+  public static Object convertToDefinedType(Object value, VCFHeaderLineType type) {
+    if (!(value instanceof List)) {
+      // deal with single value
+      // There are some field value is a single String with ',', eg: "HQ" ->"23,27"
```

Comment:
- maybe reword this comment for clarity, 
e.g. Some field values are a single, comma-separated String (e.g. "HQ" -> "23,27"

- Google Java style question: I'm not sure if multiple line comments should be placed into a block comment like 
/** text 
text 
text **/ 
or left as single line comments // @tneymanov